astropy/io/fits/fitstime.py

```diff
@@ -312,7 +415,17 @@ def fits_to_time(hdr, table):
                 if _verify_column_info(column_info, global_info):
                     colname = table.colnames[idx - 1]
                     table[colname] = _convert_time_column(table[colname],
-                                                          column_info)
+                                                          column_info,
+                                                          global_info)
+
+        other_columns = [idx for idx in range(1, len(table.columns) + 1) if
```

Comment:
I don't think it is a good idea to do this as an "afterthought". How can we even be sure that we get here (e.g., if there was no global reference time given). Instead, I  think the test should be done before the `len(hcopy)` comparison.